src/custom-elements/src/components/dropdown/dropdown.tsx

```diff
@@ -320,18 +323,24 @@ export class Dropdown {
     const menuItems = this._getDropdownMenuItems();
 
     document.body.addEventListener('click', this.handlerClick);
-    menuItems.forEach((item: HTMLElement) => {
-      item.addEventListener('click', this.handlerSelectedMenuItem);
-    });
+
+    if (!this.disableItemSelectedEvent) {
+      menuItems.forEach((item: HTMLElement) => {
+        item.addEventListener('click', this.handlerSelectedMenuItem);
+      });
```

Comment:
Same in `_removeEventListeners()`

```suggestion
    if (this.preventItemSelectedEvent) return;
    
    menuItems.forEach((item: HTMLElement) => {
       item.addEventListener('click', this.handlerSelectedMenuItem);
    });
```